#This patch file is from Qualcomm Innovation Center, Inc. and is provided under the following license:
#
#Copyright (c) 2025 Qualcomm Innovation Center, Inc. All rights reserved.
#SPDX-License-Identifier: BSD-3-Clause-Clear
From 131f3adbfaf4bc763c7a01ca57c91c81dd5bcfac Mon Sep 17 00:00:00 2001
From: Jiaxing Shi <quic_jiaxshi@quicinc.com>
Date: Mon, 24 Feb 2025 14:34:16 +0800
Subject: [PATCH] {jazzy}: enable navigation2 and nav2-rviz-plugins

Remove navigation2 and nav2-rviz-plugins from blacklist.

Upstream-Status: Pending

Signed-off-by: Jiaxing Shi <quic_jiaxshi@quicinc.com>
---
 .../ros-distro/include/jazzy/ros-distro-recipe-blacklist.inc    | 2 --
 1 file changed, 2 deletions(-)

diff --git a/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro-recipe-blacklist.inc b/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro-recipe-blacklist.inc
index d42023e9a..ef07010f8 100644
--- a/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro-recipe-blacklist.inc
+++ b/meta-ros2-jazzy/conf/ros-distro/include/jazzy/ros-distro-recipe-blacklist.inc
@@ -55,9 +55,7 @@ SKIP_RECIPE[moveit-runtime] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS'
 SKIP_RECIPE[moveit-visual-tools] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['opengl', 'x11'], 'opengl: depends on rviz-visual-tools which depends on rviz-ogre-vendor which depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES; x11: depends on rviz-visual-tools which depends on rviz-ogre-vendor which depends on libx11,libxrandr,libxaw which require x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[mrpt2] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['ffmpeg', 'openni', 'x264', 'x11'], 'ffmpeg: Depends on ffmpeg which requires commercial license; openni: Depends on libopenni2-dev which is not provided; x264: Depends on ffmpeg which depends on x264 which requires commercial license; x11: Depends on libxxf86vm, freeglut, libxrandr, wxwidgets, glfw which require x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[nav2-bringup] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: depends on slam-toolbox which depends on qtbase; x11: depends on slam-toolbox which depends on rviz-common->rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
-SKIP_RECIPE[nav2-rviz-plugins] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: depends on qtbase; x11: depends on rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[nav2-system-tests] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: depends on nav2-bringup which depends on slam-toolbox which depends on qtbase; x11: depends on nav2-bringup which depends on slam-toolbox which depends on rviz-common->rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
-SKIP_RECIPE[navigation2] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: rdepends on nav2-rviz-plugins which depends on qtbase; x11: rdepends on nav2-rviz-plugins which depends on rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[octomap-rviz-plugins] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'x11'], 'qt5: depends on qtbase; x11: depends on rviz-ogre-vendor->(libx11,libxrandr,libxaw) which require x11 in DISTRO_FEATURES', '', d)}"
 SKIP_RECIPE[octovis] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['qt5', 'libqglviewer'], 'qt5: depends on qtbase which requires meta-qt5 to be included; libqglviewer: depends on libqglviewer2-qt5 which is not available', '', d)}"
 SKIP_RECIPE[openvslam] ?= "${@bb.utils.contains_any('ROS_WORLD_SKIP_GROUPS', ['opengl', 'x11'], 'opengl: depends on libg2o which depends on mesa which is not available because of missing opengl or vulkan in DISTRO_FEATURES; x11: depends on libg2o which depends on libglu which requires x11 in DISTRO_FEATURES', '', d)}"
-- 
2.34.1

